### PR TITLE
Fix autoresume with slashed directory

### PR DESCRIPTION
**Pull request by rishab-partha**
# What does this PR do?

Fix autoresume with a slashed address for the latest filename, which ended up being that the temporary file directory was not created before trying to write to the temp file.

Output of manual test fine-tuning GPT-125M for 10 batches and then autoresuming for another 10 batches: [Wandb link to graph](https://wandb.ai/mosaic-ml/rishab-mosaicml-tests/reports/metrics-train-LanguagePerplexity-23-06-06-20-47-44---Vmlldzo0NTc4NDk0)

# What issue(s) does this change relate to?

[CO-1991](https://mosaicml.atlassian.net/browse/CO-1991
)

[CO-1991]: https://mosaicml.atlassian.net/browse/CO-1991?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: composer/utils/object_store/libcloud_object_store.py [hunk 1]
```diff
@@ -179,6 +182,7 @@ def download_object(
                 for chunk in iterate_with_callback(stream, obj.size, callback):
                     f.write(chunk)
         except Exception as e:
+            print('yup')
```
Comment (dakinggg) [inline composer/utils/object_store/libcloud_object_store.py:185]: remove

File: composer/loggers/remote_uploader_downloader.py [hunk 2]
```diff
@@ -438,7 +438,8 @@ def download_file(
                  destination=destination,
                  object_store=self.remote_backend,
                  overwrite=overwrite,
-                 progress_bar=progress_bar)
+                 progress_bar=progress_bar,
+                 marker=self.remote_backend.__class__.__name__ + 'bar')
```
Comment (dakinggg) [inline composer/loggers/remote_uploader_downloader.py:442]: what is all of this marker stuff? was it just for debugging?

**Review by dakinggg — COMMENTED** [review]
Could you please include in the pr description the output of a manual test using this (e.g. a successful autoresumption with a slashes latest filename)?

File: composer/utils/file_helpers.py [hunk 3]
```diff
@@ -415,13 +415,11 @@ def maybe_create_remote_uploader_downloader_from_uri(
...
 
 
-def get_file(
-    path: str,
-    destination: str,
-    object_store: Optional[Union[ObjectStore, LoggerDestination]] = None,
-    overwrite: bool = False,
-    progress_bar: bool = True,
-):
+def get_file(path: str,
```
Comment (dakinggg) [inline composer/utils/file_helpers.py:418]: tiny nit, can you add the trailing commas back to these so the formatting doesn't change?

**Review by dakinggg — COMMENTED** [review]
lgtm, leaving approval for evan

File: composer/utils/file_helpers.py (hunk 3 above)
Comment (rishab-partha) [reply composer/utils/file_helpers.py:418]: fixed!

**Review by eracah — APPROVED** [review]
Amazing well done! Thanks for fixing this!

**Pull request merged**